File: mobile/app/src/main/java/dam/isi/frsf/utn/edu/ar/delivery/model/Flavor.java
package dam.isi.frsf.utn.edu.ar.delivery.model;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

import java.io.Serializable;

import dam.isi.frsf.utn.edu.ar.delivery.constants.appConstants;

public class Flavor implements Serializable {

    private final static long serialVersionUID = -5244957648397091990L;
    @SerializedName("label")
    @Expose
    private String label;
    @SerializedName("imgURL")
    @Expose
    private String imgURL;

    public String getLabel() {
        return label;
    }

    public void setLabel(String label) {
        this.label = label;
    }

    public Flavor withLabel(String label) {
        this.label = label;
        return this;
    }

    public String getImgURL() {
        return imgURL;
    }

    public void setImgURL(String imgURL) {
        this.imgURL = imgURL;
    }

    public Flavor withImgURL(String imgURL) {
        this.imgURL = imgURL;
        return this;
    }

    public String getCompleteImgURL() {
        return appConstants.staticPath + "flavorImages/" + imgURL;
    }
}

File: mobile/app/src/main/java/dam/isi/frsf/utn/edu/ar/delivery/model/Flavor.kt
package dam.isi.frsf.utn.edu.ar.delivery.model

import android.os.Parcel
import android.os.Parcelable
import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName
import dam.isi.frsf.utn.edu.ar.delivery.constants.appConstants

data class Flavor(
        @SerializedName("label")
        @Expose
        var label: String = "",
        @SerializedName("imgURL")
        @Expose
        var imgURL: String = ""
) : Parcelable {
    fun withLabel(label: String): Flavor {
        this.label = label
        return this
    }

    fun withImgURL(imgURL: String): Flavor {
        this.imgURL = imgURL
        return this
    }

    val completeImgURL: String = appConstants.staticPath + "flavorImages/" + imgURL

    override fun toString(): String {
        return label
    }

    companion object {
        @JvmField val CREATOR: Parcelable.Creator<Flavor> = object : Parcelable.Creator<Flavor> {
            override fun createFromParcel(source: Parcel): Flavor = Flavor(source)
            override fun newArray(size: Int): Array<Flavor?> = arrayOfNulls(size)
        }
    }

    constructor(source: Parcel) : this(
            source.readString(),
            source.readString()
    )

    override fun describeContents() = 0

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeString(label)
        dest.writeString(imgURL)
    }
}